R/plots.R

```diff
@@ -535,3 +544,190 @@ cumulative_value.plot <- function(data,
+    print(summary(lifted))
+  }
+
+  PowerPlot_graph <- ggplot(PowerPlot, aes(x = EffectSize, y = power)) +
+    geom_smooth(formula = y ~ x, method = "loess", se = FALSE, aes(colour = "Smoothed Values")) +
+    geom_line(size = 0.62, alpha = 0.8, aes(colour = "Actual Values")) +
+    scale_x_continuous(sec.axis = sec_axis( ~ . * CostPerLift ,
+                                            breaks = scales::pretty_breaks(n = breaks_x_axis),
+                                            name = "Estimated Investment"),
+                       breaks = scales::pretty_breaks(n = breaks_x_axis)) +
+    ylim(0, 1) +
+    geom_hline(yintercept = 0.8, linetype = "dashed", color = "grey") +
+    scale_colour_manual(name="Power Curve", values=c("gray80", "#52854C")) +
+    labs( x = "Effect Size", y = "Power") +
+    theme_minimal() +
+    theme(plot.title = element_text(hjust = 0.5))
+
+  suppressMessages(gridExtra::grid.arrange(
+    plot(lifted, notes = paste(
+      # "Locations:", Market$location,
+      # "\n Rank:", Market$rank,
+      # "\n Treatment Periods:", Market$duration,
+      # "\n Effect Size: ", Market$EffectSize
+    )),
+    PowerPlot_graph,
+    ncol = 1,
+    nrow = 2,
+    bottom = paste(
+      "Rank:", Market$rank,
+      "\n Locations:", Market$location,
+      "\n Treatment Periods:", Market$duration,
+      "\n Effect Size: ", Market$EffectSize
+    )))
+
+}
+
+
+#' Plotting function for historical correlations between the test market's
+#' and total markets KPI.
+#'
+#' @description
+#'
+#' `plotCorrels` takes a data frame used for GeoLift to generate
+#' a plot that show historical similarities in KPI levels between the
+#' test markets and the aggregation of all locations (total).
+#'
+#' @param data A data.frame containing the historical conversions by
+#' geographic unit. It requires a "locations" column with the geo name,
+#' a "Y" column with the outcome data (units), a time column with the indicator
+#' of the time period (starting at 1), and covariates.
+#' @param locs List of markets to use in the calculation of the correlations.
+#' @param scaled A logic flag indicating whether to plot the scaled values of
+#' the KPI metric (standardized by the largest historical value). Set to FALSE
+#' by default to plot the observed KPI levels.
+#' @param KPI_id Outcome variable.
+#' @param dtw Emphasis on Dynamic Time Warping (DTW), dtw = 1 focuses exclusively
+#' on this metric while dtw = 0 (default) relies on correlations only.
+#'
+#' @return
+#' A plot of the historical values of the test market and the aggregation of
+#' all control markets.
+#'
+#' @export
+plotCorrels <- function(data,
+                        locs = c(),
+                        scaled = TRUE,
+                        KPI_id = "",
+                        dtw = 0){
+
+  if(!(all(tolower(locs) %in% tolower(unique(data$location))))){
+    warning("Please specify a valid set of test locations.")
+    return(NULL)
+  }
```

Comment:
This check will already be done by AppendAgg, I would remove it.

Reply:
You're right! Removing it